src/Request.hs

```diff
@@ -77,7 +77,7 @@ mkValidator contractInfo@ContractInfo{..} dat lenderTn ctx = validate
     valueToBorrower = valuePaidTo (U.info ctx) (unPaymentPubKeyHash $ borrowersPkh dat)
 
     borrowerGetsWhatHeWants :: Bool
-    borrowerGetsWhatHeWants = assetClassValueOf valueToBorrower (loan dat) == loanamnt dat
+    borrowerGetsWhatHeWants = assetClassValueOf valueToBorrower (loan dat) >= loanamnt dat && assetClassValueOf valueToBorrower (loan dat) >= assetClassValueOf (U.valueIn ctx) (loan dat)
```

Comment:
I think this doesn't do the job for every possible loan asset class. For simplicity, let's just imagine loan currency being ADA. You can't require all ADA from all inputs being given to borrower. Aside from paying for tx fees, you also need to insert min-ADA amount to the ongoing Collateral script.

I would suggest explicitly checking that no other script input is present in the transaction (ideally, all transactions).